hcloud/client_handler_debug_test.go

```diff
@@ -0,0 +1,103 @@
+package hcloud
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDebugHandler(t *testing.T) {
+	testCases := []struct {
+		name    string
+		wrapped func(req *http.Request, v any) (*Response, error)
+		want    string
+	}{
+		{
+			name: "network error",
+			wrapped: func(_ *http.Request, _ any) (*Response, error) {
+				return nil, fmt.Errorf("network error")
+			},
+			want: `--- Request:
+GET /v1/ HTTP/1.1
+Host: api.hetzner.cloud
+User-Agent: hcloud-go/testing
+Authorization: REDACTED
```

Comment:
Can you add a test case that verifies the redaction with an actual header in the request?

Reply:
Missed the actual client + token in the test function :facepalm: